DatabaseStructure.sql

```diff
@@ -23,43 +23,44 @@ CREATE TABLE `Roles` (
 );
 
 -- Table Event
-DROP TABLE IF EXISTS `Event`;
 CREATE TABLE `Event` (
    event_id INT AUTO_INCREMENT NOT NULL,
    event_name VARCHAR(70) UNIQUE NOT NULL,
    event_description TEXT NOT NULL,
    location_street VARCHAR(50),
    location_postal_code VARCHAR(6),
    location_city VARCHAR(30),
-   claimed BOOLEAN,
+   event_time TIME,
+   event_date DATE,
+   inactive BOOLEAN,
    PRIMARY KEY(event_id)
 );
 
 -- Table User
-DROP TABLE IF EXISTS `User`;
+
 CREATE TABLE `User` (
    user_id INT AUTO_INCREMENT NOT NULL,
-   user_name VARCHAR(25) UNIQUE NOT NULL,
    user_password VARCHAR(60) NOT NULL,
    password_change_date TIMESTAMP,
    first_name VARCHAR(25) NOT NULL,
    last_name VARCHAR(25) NOT NULL,
    email_address VARCHAR(50) UNIQUE NOT NULL,
    type_of_staff INT NOT NULL,
    user_role INT NOT NULL,
+   inactive BOOLEAN,
    PRIMARY KEY(user_id),
-   FOREIGN KEY(type_of_staff) REFERENCES TypesOfStaff(type_of_staff_id) ON UPDATE CASCADE,
-   FOREIGN KEY(user_role) REFERENCES Roles(role_id) ON UPDATE CASCADE
+   FOREIGN KEY(type_of_staff) REFERENCES TypesOfStaff(type_of_staff_id) ON UPDATE CASCADE ON DELETE NO ACTION,
+   FOREIGN KEY(user_role) REFERENCES Roles(role_id) ON UPDATE CASCADE ON DELETE NO ACTION
 );
 
 -- Table Event_Details
-DROP TABLE IF EXISTS `Event_Details`;
 CREATE TABLE `Event_Details` (
```

Comment:
Needs a way to check whether a user is currently checked or has checked out already in (check out/check in variable in database).

Reply:
Check in will be stated by creating the entry into the "event_details" table and checkout by deleting that entry